common/builder/pystache_message_builder.py

```diff
@@ -20,4 +26,16 @@ def build_message(self, message_dictionary):
         :param message_dictionary: The dictionary of values to use when populating the template.
         :return: A string containing a message suitable for sending to a remote MHS.
         """
-        return self._renderer.render(self._parsed_template, message_dictionary)
+        try:
+            return self._renderer.render(self._parsed_template, message_dictionary)
+        except KeyNotFoundError as e:
+            logger.error('0001', 'Failed to find key when generating message from {TemplateFile} . {ErrorMessage}',
+                         {'TemplateFile': self.template_file, 'ErrorMessage': str(e)})
+            raise MessageGenerationSOAPFaultError()
+
+
+class MessageGenerationSOAPFaultError(Exception):
```

Comment:
Thinking of changing this to something like `MessageGenerationError` or some more specific thing like `MessageGenerationMissingTagError`. The reason I bunged `SOAPFault` in the name originally was because it was mentioned in [Confluence](https://gpitbjss.atlassian.net/wiki/spaces/RTDel/pages/1562149077/ebXML+SOAP+MessageHeader+Ack+to+Spine?atlOrigin=eyJpIjoiODg4MDEwOTk3MDYxNGI2Y2FmZmU5YzBmNjRjMjIwOTkiLCJwIjoiYyJ9).